mlflow/utils/name_utils.py

```diff
@@ -0,0 +1,305 @@
+import random
+
+
+def _generate_random_name(sep="-", integer_scale=3):
+    """Helper function for
+
+    :param sep: String seperator for word spacing
+    :param integer_scale: dictates the maximum scale range for random integer sampling (power of 10)
+
+    :return: A random string phrase comprised of a predicate, noun, and random integer
+    """
+
+    predicate = random.choice(_GENERATOR_PREDICATES).lower()
+    noun = random.choice(_GENERATOR_NOUNS).lower()
+    num = random.randint(0, 10**integer_scale)
```

Comment:
I like these words. There aren't too too many of them, so we don't risk adding an excessive amount of text content to the MLflow library, and it's pretty easy to extend them later. We can generate ~ 20k unique pairings of predicate / animal and 20 million unique run names, which seems sufficiently distinct :D

Reply:
@BenWilson2 @dbczumar - I am not able to understand how we are guarantying no name collisions by just relying on random. Is that handled in some other code flow that I have missed or there is there something that I am missing here.

Reply:
Run name isn't a primary key. Why would a collision-free solution be required here?

Reply:
Thanks @BenWilson2 . Yes you are correct, i think i got confused that name is a primary key as duplicate experiment name was not allowed in mlflow. It is cleared now.